src/ray/object_manager/object_manager.cc

```diff
@@ -378,39 +378,51 @@ void ObjectManager::Push(const ObjectID &object_id, const NodeID &node_id) {
 
 void ObjectManager::PushLocalObject(const ObjectID &object_id, const NodeID &node_id) {
   const ObjectInfo &object_info = local_objects_[object_id].object_info;
-  uint64_t total_data_size =
-      static_cast<uint64_t>(object_info.data_size + object_info.metadata_size);
+  uint64_t data_size = static_cast<uint64_t>(object_info.data_size);
   uint64_t metadata_size = static_cast<uint64_t>(object_info.metadata_size);
-  uint64_t num_chunks = buffer_pool_.GetNumChunks(total_data_size);
 
   rpc::Address owner_address;
   owner_address.set_raylet_id(object_info.owner_raylet_id.Binary());
   owner_address.set_ip_address(object_info.owner_ip_address);
   owner_address.set_port(object_info.owner_port);
   owner_address.set_worker_id(object_info.owner_worker_id.Binary());
 
-  auto local_chunk_reader = [this, object_id, total_data_size, metadata_size](
-                                uint64_t chunk_index,
-                                rpc::PushRequest &push_request) -> Status {
-    std::pair<const ObjectBufferPool::ChunkInfo, ray::Status> chunk_status =
-        buffer_pool_.GetChunk(object_id, total_data_size, metadata_size, chunk_index);
-    // Fail on status not okay. The object is local, and there is
-    // no other anticipated error here.
-    Status status = chunk_status.second;
-    if (status.ok()) {
-      ObjectBufferPool::ChunkInfo chunk_info = chunk_status.first;
-      push_request.set_data(chunk_info.data, chunk_info.buffer_length);
-    }
-    return status;
-  };
+  std::pair<std::shared_ptr<MemoryObjectReader>, ray::Status> reader_status =
+      buffer_pool_.CreateObjectReader(object_id, owner_address);
+  Status status = reader_status.second;
+  if (!status.ok()) {
```

Comment:
@rkooo567 one behavior change here is that if the object read failed the push manager won't send anything over the network. will this cause any issue?

Reply:
What did it send over the network before?